integration_tests/test_get_requests.py

```diff
@@ -1,96 +1,47 @@
-import requests
+from requests import Response
 
-BASE_URL = "http://127.0.0.1:8080"
+from http_methods_helpers import get
 
 
-def test_index_request(session):
-    res = requests.get(f"{BASE_URL}")
-    assert res.status_code == 200
+def test_param(session):
+    r = get("/sync/param/1")
+    assert r.text == "1"
+    r = get("/sync/param/12345")
+    assert r.text == "12345"
+    r = get("/async/param/1")
+    assert r.text == "1"
+    r = get("/async/param/12345")
+    assert r.text == "12345"
 
 
-def test_jsonify(session):
-    r = requests.get(f"{BASE_URL}/jsonify")
-    assert r.json() == {"hello": "world"}
-    assert r.status_code == 200
+def test_serve_html(session):
+    def check_response(r: Response):
+        assert r.text.startswith("<!DOCTYPE html>")
+        assert "Hello world. How are you?" in r.text
 
+    check_response(get("/sync/serve/html"))
+    check_response(get("/async/serve/html"))
 
-def test_html(session):
-    r = requests.get(f"{BASE_URL}/test/123")
-    assert "Hello world. How are you?" in r.text
 
+def test_template(session):
+    def check_response(r: Response):
+        assert r.text.startswith("\n\n<!DOCTYPE html>")
+        assert "Jinja2" in r.text
+        assert "Robyn" in r.text
 
-def test_jinja_template(session):
-    r = requests.get(f"{BASE_URL}/template_render")
-    assert "Jinja2" in r.text
-    assert "Robyn" in r.text
+    check_response(get("/sync/template"))
```

Comment:
One last nit @AntoineRR - let's split the async and the sync tests in two different parts? I know this will make the test suite a bit slower but it will make things easier to test.